tests/core/filtering/test_contract_data_filters.py

```diff
@@ -162,37 +177,45 @@ def test_data_filters_with_list_arguments(
         emitter_event_ids,
         call_as_instance,
         create_filter,
+        api_style,
         vals):
 
     matching, non_matching = vals
 
     if call_as_instance:
-        event_filter = create_filter(emitter, [
-            'LogListArgs', {
-                'filter': {"arg1": matching}}])
+        contract = emitter
```

Comment:
Somewhat orthogonal but this could be to use the [`request`](https://docs.pytest.org/en/latest/reference.html#_pytest.fixtures.FixtureRequest.getfixturevalue) fixture to lazily request these fixtures:

```python
def the_test(request):
    if call_as_instance:
        contract = request.getfixturevalue('emitter')
    else:
        contract = request.getfixturevalue('Emitter')
```

It'd give us *some* increase in test speed since it won't need to deploy the `emitter` contract for every single run.